File: frontend/tests/utils/mock/types/fft.ts
import type { FftNode } from '../../../../src/api/types.gql-generated';

export type FFT = Omit<FftNode, 'spectrogramAnalysis'>
export const fft: FFT = {
  id: '1',
  samplingFrequency: 480,
  overlap: 0.50,
  nfft: 2048,
  windowSize: 2048,
  legacy: true,
}

File: backend/api/admin/data/fft.py
"""API data fft administration"""
from django.contrib import admin

from backend.api.models import FFT


@admin.register(FFT)
class FFTAdmin(admin.ModelAdmin):
    """FFT presentation in DjangoAdmin"""

    # pylint: disable=duplicate-code
    list_display = (
        "id",
        "nfft",
        "window_size",
        "overlap",
        "sampling_frequency",
        "scaling",
        "legacy",
    )
